File: EDAII x Python/Ordenamiento/BitonicSort.py
import math

class BitonicSort:

	arr = []

	def __init__(self,arr):
		'Constructor de la clase'
		self.arr = arr

	def bitonicSort(self,arr,inicio,tam,dir):
		'Algoritmo de Ordenamiento'
		if tam > 1:
			k = math.floor(tam/2)

			# Ordena lado izquierdo de forma ascendente (menor-mayor)
			self.bitonicSort(arr,inicio,k,True)

			# Ordena lado derecho de forma descendente (mayor-menor)
			self.bitonicSort(arr,inicio+k,k,False)

			# Mezcla dos mitades en determinado orden
			self.bitonicMerge(arr,inicio,tam,dir)

	def bitonicMerge(self,arr,inicio,tam,dir):
		'Mezcla dos subarreglos de tamaño 2^n'
		if tam > 1:
			k = math.floor(tam/2)

			for i in range(inicio,inicio+k):
				# Intercambia dependiendo el orden asignado
				if (arr[i]>arr[i+k] and dir==True) or (arr[i]<arr[i+k] and dir==False):
					arr[i],arr[i+k] = arr[i+k],arr[i]

			self.bitonicMerge(arr,inicio,k,dir)
			self.bitonicMerge(arr,inicio+k,k,dir)

# Ejecutando el objeto
# Nota: BitonicSort solo funciona con arreglos de longitud 2^n
programa = BitonicSort([10, 24, 2, 10, 11, 27, 4, 2])
programa.bitonicSort(programa.arr,0,len(programa.arr),True)
print(programa.arr)

File: EDAII x Java/Ordenamiento/BitonicSort.java
public class BitonicSort{

	// Nota: BitonicSort solo funciona con arreglos de longitud 2^n
	int[] arr = {10, 24, 2, 10, 11, 27, 4, 2};

	public static void main(String[] args){
		BitonicSort bs = new BitonicSort();
		bs.bitonicSort(bs.arr, 0, bs.arr.length, true);

		for (int i=0; i<bs.arr.length; i++){
			System.out.print(bs.arr[i] + " ");
		}
		System.out.println("\n");
	}

	public void bitonicSort(int[] A, int inicio, int tam, boolean dir){
		if (tam > 1){
			int k = tam/2;

			// Ordena lado izquierdo de forma ascendente (menor-mayor)
			bitonicSort(A, inicio, k, true);

			// Ordena lado derecho de forma descendente (mayor-menor)
			bitonicSort(A, inicio+k, k, false);

			// Mezcla dos mitades en determinado orden
			bitonicMerge(A, inicio, tam, dir);
		}
	}

	public void bitonicMerge(int[] A, int inicio, int tam, boolean dir){
		if (tam > 1){
			int k=tam/2;

			for (int i=inicio; i<inicio+k; i++){
				// Intercambia dependiendo el orden asignado
				if ((A[i]>A[i+k] && dir==true) || (A[i]<A[i+k] && dir==false)){
					int x = A[i];
					A[i] = A[i+k];
					A[i+k] = x;
				}
			}
			bitonicMerge(A,inicio,k,dir);
			bitonicMerge(A,inicio+k,k,dir);
		}
	}

}